ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/installFeatureWithAuth.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.metrics.micrometer.prometheus
+
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.routing.*
+
+fun Application.installMetricsEndpointWithAuth(config: MicrometerMetricsPrometheus.Configuration) {
+
+    checkIfAuthenticationFeatureIsInstalled()
+
+    routing {
+        authenticate(*config.authenticationConfig) {
+            metricsEndpoint(config)
+        }
+    }
+}
+
+private fun Application.checkIfAuthenticationFeatureIsInstalled() {
+    try {
```

Comment:
According to your description, auth is optional however we actually always require auth.

Reply:
only if authentication() config is set in the configuration, if this is not set, we dont need the auth feature

Reply:
Also, I would need, if possible, to be able to install the endpoint in a specific port (I have an internal port for prometheus scraping, and an external one that our proxy service exposes to the internet).

Reply:
Maybe like providing an `var endpointPort: Int? = null`?

Reply:
hm... the next request would to specify the host name and now we start to reimplement the routing declaration. 
Alternative to the current implementation would be: The feature itself declares by default the endpoint under `/metrics` without any authentication, port selection etc. If you need some special routing feature you set a property `declareEndpoint = false` in the config, then you can declare the endpoint like you want it to be:
```
routing {
   port(1234) {
      authentication("superSecret") {
         //...
         get("/myMetricsPath") {
            prometheusMetrics()
          }
    }
}

Reply:
The other solution is to provide routing entry for prometheus endpoint without any built-in auth. Like this:

```
fun Routing.prometheusEndpoint(...) { ... }

// so one can install it anywhere 
app.routing {
    route("/api") {
        prometheus { enable public metrics }
    }
    route("/admin") {
        authenticate(...) {
            // some admin actions go here
            prometheus { enable all metrics }
        }
    }
}
```

Reply:
The advantage is that you don't need to care of auth at all. This is much more composable.